test/catch2_test_block_shuffle.cu

```diff
@@ -175,45 +195,55 @@ struct down_with_prefix_op_t
 
     block_shuffle.Down(thread_data, thread_data, prefix);
 
-    if (static_cast<int>(threadIdx.x) == m_target_thread_id)
+    if (static_cast<int>(threadIdx.x) == m_target_thread_id_x
+     && static_cast<int>(threadIdx.y) == m_target_thread_id_y
+     && static_cast<int>(threadIdx.z) == m_target_thread_id_z)
     {
       m_d_prefix_ptr[0] = prefix;
     }
   }
 };
 
 template <int ItemsPerThread,
-          int ThreadsInBlock,
+          int BlockDimX,
+          int BlockDimY,
+          int BlockDimZ,
           class T,
           class ActionT>
 void block_shuffle(
     thrust::device_vector<T> &data, 
     ActionT action)
 {
-  block_shuffle_kernel<ThreadsInBlock, ItemsPerThread>
-    <<<1, ThreadsInBlock>>>(
+  dim3 block(BlockDimX, BlockDimY, BlockDimZ);
+  block_shuffle_kernel<BlockDimX, BlockDimY, BlockDimZ, ItemsPerThread>
+    <<<1, block>>>(
         thrust::raw_pointer_cast(data.data()),
         action);
 
   REQUIRE( cudaSuccess == cudaPeekAtLastError() );
   REQUIRE( cudaSuccess == cudaDeviceSynchronize() );
 }
 
-// %PARAM% THREADS_IN_BLOCK bs 64:512:1024
+// %PARAM% TEST_DIM_X dimx 7:32:64:128
```

Comment:
I don't think we need to test the whole Cartesian product for multi-dimentional thread blocks. Let's have a single test for a few block sizes and keep large thread blocks covered in the rest of the tests.